packages/gatsby-theme-carbon/gatsby-config.mjs

```diff
@@ -9,6 +9,16 @@ import remarkGfm from 'remark-gfm';
 import { fileURLToPath } from 'url';
 import defaultLunrOptions from './config/lunr-options.mjs';
 
+/*
+  This is a rehype plugin that adds support for metadata to the fenced code block
+  For eg: 
+  ```jsx path=/directory/file.mdx src=https://gatsby.carbondesignsystem.com
+  const a = 10;
+  ```
+  A metaData prop of format path=/directory/file.mdx src=https://gatsby.carbondesignsystem.com is added to the code block
+*/
+import rehypeAddCodeMetaData from 'rehype-mdx-fenced-code-meta-support';
```

Comment:
Created a rehype package which is adding a new prop of named metaData which is being passed in Code.js.

Here is the dirty package - https://www.npmjs.com/package/rehype-mdx-fenced-code-meta-support